pycbc/events/coinc.py

```diff
@@ -91,34 +96,34 @@ def background_bin_from_string(background_bins, data):
             elif bin_type == 'chi_eff':
                 vals = pycbc.conversions.chi_eff(data['mass1'], data['mass2'],
                                                  data['spin1z'], data['spin2z'])
-            elif bin_type == 'SEOBNRv2Peak':
-                vals = pycbc.pnutils.get_freq('fSEOBNRv2Peak',
-                                              data['mass1'], data['mass2'],
-                                              data['spin1z'], data['spin2z'])
-            elif bin_type == 'SEOBNRv4Peak':
-                vals = pycbc.pnutils.get_freq('fSEOBNRv4Peak', data['mass1'],
-                                              data['mass2'], data['spin1z'],
-                                              data['spin2z'])
-            elif bin_type == 'SEOBNRv2duration':
-                vals = pycbc.pnutils.get_imr_duration(
-                                   data['mass1'], data['mass2'],
-                                   data['spin1z'], data['spin2z'],
-                                   data['f_lower'], approximant='SEOBNRv2')
-            elif bin_type == 'SEOBNRv4duration':
-                vals = pycbc.pnutils.get_imr_duration(
-                                   data['mass1'][:], data['mass2'][:],
-                                   data['spin1z'][:], data['spin2z'][:],
-                                   data['f_lower'][:], approximant='SEOBNRv4')
-            elif bin_type == 'SEOBNRv5duration':
+            elif bin_type in ['SEOBNRv2Peak', 'SEOBNRv4Peak']:
+                vals = pycbc.pnutils.get_freq(
+                    'f' + bin_type,
+                    data['mass1'],
+                    data['mass2'],
+                    data['spin1z'],
+                    data['spin2z']
+                )
+                cached_values[bin_type] = vals
+            elif bin_type in ['SEOBNRv2duration', 'SEOBNRv4duration', 'SEOBNRv5duration']:
+                approx_map = {
+                    'SEOBNRv2duration': 'SEOBNRv2',
+                    'SEOBNRv4duration': 'SEOBNRv4',
+                    'SEOBNRv5duration': 'SEOBNRv5_ROM'
+                }
```

Comment:
If you define `approx_map` outside of this `if` block, then you can just do
```
elif bin_type in approx_map
```
and avoid some repetition.

Reply:
Good point, I've changed this.